### ISSUE TITLE
validate-website-static --not-found fails with anchor links

### ISSUE THREAD

**Issue by tkrotoff** [spk/validate-website#14]
Example of an anchor link: `<a href="/tags/#ostéopathie">#ostéopathie</a>`

Everything works fine using validate-website:
```
$ bundle exec validate-website --site 'https://osteo15.com' --verbose --markup --not-found
validating https://osteo15.com

.............................
```

Fails with validate-website-static:
```
$ bundle exec validate-website-static --site 'https://osteo15.com' --verbose --markup --not-found
validating https://osteo15.com

.
/Users/tanguy/osteo15.com/_site/tags/%23ost%C3%A9opathie linked but not exist

/Users/tanguy/osteo15.com/_site/tags/%23FemmeEnceinte linked but not exist

/Users/tanguy/osteo15.com/_site/tags/%23interview linked but not exist
```

Caused by bad escaping?

FYI
- List of valid URL characters: [`ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~:/?#[]@!$&'()*+,;=`](http://stackoverflow.com/a/1547940/990356)
- List of invalid filename characters under Windows: [`\/:*?"<>|`](https://msdn.microsoft.com/en-us/library/aa365247)
- List of invalid filename characters under UNIX (depends on the filesystem): [`NUL (\0)` and `/`](http://en.wikipedia.org/wiki/Ext4)

**Comment by spk** [comment]
Thanks for your report :exclamation:
I will check that.

**Comment by spk** [comment]
Released v1.0.3

**Comment by tkrotoff** [comment]
Super, merci
